[Tan Jun Heng] iP

Austintjh19 opened the pull request:
# Osiris - A Better Duke. 

> _“Your mind is for having ideas, not holding them.”_ – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Osiris frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

--- 

If you are a Java programmer, you can use it to practice Java too. Here's the  `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Austintjh19 committed:
Commit Level 0. Rename, Greet, Exit.

Austintjh19 committed:
Adding Level-0 Tag.

Austintjh19 committed:
Commit Level 0. Rename, Greet, Exit.

Austintjh19 committed:
Commit Level 0. Rename, Greet, Exit.

Austintjh19 committed:
Commit Level 0. Rename, Greet, Exit.

Austintjh19 committed:
Commit Level 0. Rename, Greet, Exit.

Austintjh19 committed:
Commit Level 0. Rename, Greet, Exit.

Austintjh19 committed:
Level 1. Echo. { Added New Class ChatInterface. }

Austintjh19 committed:
Level 1. Echo. { Added New Class ChatInterface. }

Austintjh19 committed:
Level 1. Echo. { Removed chat interface class }

Austintjh19 committed:
Level 1. Echo. { Removed chat interface class }

Austintjh19 committed:
Merge branch 'master' of https://github.com/Austintjh19/ip

Austintjh19 committed:
Level 1. Echo. { Removed chat interface class }

Austintjh19 committed:
Level 2. Add, List. { UserInputs Class added. }

Austintjh19 committed:
Level 3. Mark as Done. { Refractor UserInputs class to TaskManager }

Austintjh19 committed:
Level 3. Mark as Done. { Create new class Task }

Austintjh19 committed:
Level 3. Mark as Done. { Completed }

Austintjh19 committed:
Level 4. ToDos, Events, Deadlines. { Added DeadlineTask, ToDoTask & Event Task class. Modified Osiris to use case statements. Modified Osiris to handle String to Integer Exceptions }

Austintjh19 committed:
Level 4. ToDos, Events, Deadlines. { Completed }

Austintjh19 committed:
A-TextUiTesting. Level-4. { Passed }

Austintjh19 committed:
Level 5. Handle Errors. { Validation Check and Exception Handling for Mark & Unmark Task }

Austintjh19 committed:
Level 5. Handle Errors. { Completed. Incorporated validation checks for user inputs. Felt that it was more appropriate than Exceptions. However, did create a InvalidTaskException for future modifications. }

Austintjh19 committed:
Level 6. Delete. { Completed.  Validation checks and Exception Handling were incorporated. }

Austintjh19 committed:
Chatbot Introductions Modification.

Austintjh19 committed:
Level 7. Save. { Check Branch }

Austintjh19 committed:
Level 7. Save. { Branch Testing }

Austintjh19 committed:
Level 7. Save. { Branch Check }

Austintjh19 committed:
Level 7. Save. { Branch Testing Done. }

Austintjh19 committed:
Level 7. Save. { New Storage & TxtFileStorage Classes }.

Austintjh19 committed:
Level 7. Save. { Completed File Storing of Task }

Austintjh19 committed:
Level 7. Save. { Complete }

Austintjh19 committed:
Merge branch 'branch-Level-7'

Austintjh19 committed:
Level 8. Dates and Times. { Branch Test }

Austintjh19 committed:
Level 8. Dates and Times. { Branch Test }

Austintjh19 committed:
Level 8. Dates & Times. { Created Ui Class }

Austintjh19 committed:
Level 8. Dates & Times { Ui class Integrated. }

Austintjh19 committed:
Level 8. Dates and Times. { Refractor classes, created new classes InputInterpreter and Command. }

Austintjh19 committed:
Level 8. Dates and Times. { Added InputsValidator Class }

Austintjh19 committed:
Level 8. Dates and Times. { Integration of InputInterpreter, Command, & Validation Class }

Austintjh19 committed:
Level 8. Date and Times. { Task now stores LocalDate and LocalDateTime. Integrated it with TaskManager and UI. }

Austintjh19 committed:
Level 8. Dates and Times. { Added Validation for Date Time inputs. }

Austintjh19 committed:
Merge branch 'branch-Level-8'

Austintjh19 committed:
A-Packages. { Completed }

Austintjh19 committed:
Merge branch 'master' into add-gradle-support

Austintjh19 committed:
A-Gradle. { Completed }

Austintjh19 committed:
A-JUnit. { Completed Key functions }

Austintjh19 committed:
A-Jar. { Completed }.

Austintjh19 committed:
Merge branch 'add-gradle-support'

Austintjh19 committed:
A-JavaDoc.

Austintjh19 committed:
A-JavaDoc. { Completed. }

Austintjh19 committed:
A-CodingStandard. { Completed }

Austintjh19 committed:
Level 9. Find. { Completed }

Austintjh19 committed:
Merge branch 'branch-A-JavaDoc'

Austintjh19 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/osiris/Osiris.java
#	src/main/java/osiris/commands/NoCommand.java
#	src/main/java/osiris/commands/UnsupportedCommand.java
#	src/main/java/osiris/commands/addDeadlineTaskCommand.java
#	src/main/java/osiris/commands/addEventTaskCommand.java
#	src/main/java/osiris/commands/addToDoTaskCommand.java
#	src/main/java/osiris/commands/markTaskCompletedCommand.java
#	src/main/java/osiris/commands/markTaskIncompleteCommand.java
#	src/main/java/osiris/commands/printUserTasksCommand.java
#	src/main/java/osiris/commands/removeTaskCommand.java
#	src/main/java/osiris/commands/terminateChatCommand.java
#	src/main/java/osiris/task/EventTask.java
#	src/main/java/osiris/task/InvalidTaskException.java
#	src/main/java/osiris/task/Task.java
#	src/main/java/osiris/task/TaskManager.java
#	src/main/java/osiris/ui/Ui.java

Austintjh19 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/osiris/task/TaskManager.java
#	src/main/java/osiris/ui/Ui.java

Austintjh19 committed:
Coding Standard fix: Updated according to Coding Standard Provided.

Austintjh19 committed:
Coding Standard fix: Updated according to Coding Standard Provided.

Test files have Coding Standards Violation.
Updated tags for A-JavaDoc, A_CodingStandards & A-JUnit.

Austintjh19 committed:
Initialise basics for JafaFx

Austintjh19 committed:
Add Exception Classes

For the purpose of integrating with the JavaFX interface.
Println commands no longer viable after validation checks.

Austintjh19 committed:
Add JavaDoc for new classes.

Austintjh19 committed:
Update Unit Tests

Austintjh19 committed:
Merge branch 'branch-Level-10'

wang-xinrong commented on src/main/java/osiris/commands/AddDeadlineTaskCommand.java:
```diff
@@ -0,0 +1,51 @@
+/**
+ * Command class representing the addition of a deadline task.
+ */
+public class AddDeadlineTaskCommand extends Command {
```
I like this abstraction of command, great work!

wang-xinrong commented on src/main/java/osiris/commands/AddDeadlineTaskCommand.java:
```diff
@@ -0,0 +1,51 @@
+    private final LocalDate deadline;
+
+    /**
+     * Constructs an AddDeadlineTaskCommand object with the specified task name and deadline.
```
meticulous effort in documenting the public methods 👍🏼

wang-xinrong commented on src/main/java/osiris/commands/PrintUserTasksCommand.java:
```diff
@@ -0,0 +1,32 @@
+package osiris.commands;
+
+import java.util.ArrayList;
```
nice grouping of import statements 👍🏼

wang-xinrong commented on src/main/java/osiris/interpreters/UserInputInterpreter.java:
```diff
@@ -0,0 +1,139 @@
+        boolean isValid = false;
+
+        switch (inputtedWords[0]) {
+        case TerminateChatCommand.COMMAND:
```
I love how the COMMAND field is used to identify specific commands, great work 👍🏼

wang-xinrong commented on src/test/java/osiris/formatters/DateTimeFormattersTest.java:
```diff
@@ -0,0 +1,154 @@
+/**
+ * Unit tests for the DateTimeFormatters class.
+ */
+public class DateTimeFormattersTest {
```
Comprehensive test suite for DateTimeFormatter 👍🏼

wang-xinrong reviewed (commented):
LGTM, great code structure, meticulous efforts in documenting and testing. Great work overall 👍🏼

Austintjh19 committed:
Update Task Storage File Path

Austintjh19 committed:
Generate new Osiris Jar File

Austintjh19 committed:
Enhance Codebase with Assert Statements in Osiris

Implemented assert statements across the codebase to enforce data
integrity and enhance reliability.

Primarily focused on the Validation class, ensuring thorough
input validation procedures.

Austintjh19 committed:
Integrate Stream for Print Task Execution

Utilised streams and collectors to transform task objects into their
string representations.

Enhances code clarity and reduces complexity.

Austintjh19 committed:
Refactor task notification messages

Avoided and removed complicated expressions to improve readability.

Austintjh19 committed:
Improve code quality

Examined the code and ensured happy path is prominent.

Examined the code and removed dead code.

Examined the code and avoided deep nesting when possible. Balance had
to be found with ensuring the happy path is prominent and avoiding
nested conditionals.

Austintjh19 committed:
Merge pull request #1 from Austintjh19/branch-A-Streams

A-Streams

Austintjh19 committed:
Merge branch 'master' of https://github.com/Austintjh19/ip

Austintjh19 committed:
Merge branch 'master' into branch-A-CodeQuality

Austintjh19 committed:
Merge branch 'branch-A-CodeQuality' into branch-A-Assertions

Austintjh19 committed:
Merge branch 'master' into branch-A-Assertions

Austintjh19 committed:
Merge branch 'branch-A-Assertions' of https://github.com/Austintjh19/ip into branch-A-Assertions

Austintjh19 committed:
Merge pull request #2 from Austintjh19/branch-A-CodeQuality

A-CodeQuality

Austintjh19 committed:
Merge branch 'master' into branch-A-Assertions

Austintjh19 committed:
Merge pull request #3 from Austintjh19/branch-A-Assertions

A-Assertions

Austintjh19 committed:
CI Workflow Java Build and IO Redirection Tests

Added a continuous integration (CI) workflow for this project.

Austintjh19 committed:
Replace System.out with System.err for Error Messaging

By utilizing System.err, error messages are appropriately directed
to the standard error stream, which is more suitable for error output.

Austintjh19 committed:
Add sort by start of task date time feature

Implemented a feature to sort tasks by their deadlines.
The tasks are now displayed in ascending order of deadlines.

Austintjh19 committed:
Implement a more flexible search

Will now be able to find items even if the keyword matches the item
only partially.

Austintjh19 committed:
Merge branch 'branch-BCD-Extension'

Austintjh19 committed:
Improve the GUI

Modified background to make it more pleasing.
Modified text font so it is more appropriate.

Austintjh19 committed:
Merge branch 'branch-A-BetterGui'

Austintjh19 committed:
Add a unique personality

Austintjh19 committed:
Merge branch 'branch-A-Personality'

Austintjh19 committed:
Use AI to remove Arrow Head Conditionals

Austintjh19 committed:
Merge branch 'branch-A-AiAssisted'

Austintjh19 committed:
Add a representative screenshot

Austintjh19 committed:
Update Readme

Austintjh19 committed:
Add App Tittle

Austintjh19 committed:
Update Readme wit Release
